src/model/tableNode.ts

```diff
@@ -62,4 +62,31 @@ export class TableNode implements INode {
 
         Utility.runQuery(sql, connection);
     }
+
+    public async dropTable() {
+        const options: vscode.MessageOptions = {
+            modal: true,
+        };
+        const answer = await vscode.window.showWarningMessage(`Are you sure you want to drop table ${this.table}?`, options, "Drop table");
+
+        if (answer === undefined) {
+            return;
+        }
+
+        AppInsightsClient.sendEvent("drop table");
+        const sql = `DROP TABLE ${this.database}.${this.table}`;
+        Utility.createSQLTextDocument(sql);
+
+        const connection = {
+            host: this.host,
+            user: this.user,
+            password: this.password,
+            port: this.port,
+            database: this.database,
+            certPath: this.certPath,
+        };
+        Global.activeConnection = connection;
+
+        Utility.runQuery(sql, connection);

```

Comment:
How about refreshing the table list after deletion is successful?

Reply:
How to i refresh the databaseNode from inside the tableNode class? I want to call: 

mysqlTreeDataProvider.refresh(databaseNode);

How can i get the "parent" databaseNode for a tableNode?

Reply:
Like this

´public async dropTable(mysqlTreeDataProvider?: MySQLTreeDataProvider) {
        const options: vscode.MessageOptions = {
            modal: true,
        };
        const answer = await vscode.window.showWarningMessage(`Are you sure you want to drop table ${this.table}?`, options, "Drop table");

        if (answer === undefined) {
            return;
        }

        AppInsightsClient.sendEvent("dropTable");
        const sql = `DROP TABLE ${this.database}.${this.table}`;
        Utility.createSQLTextDocument(sql);

        const connection = {
            host: this.host,
            user: this.user,
            password: this.password,
            port: this.port,
            database: this.database,
            certPath: this.certPath,
        };
        Global.activeConnection = connection;

        Utility.runQuery(sql, connection);

        if (mysqlTreeDataProvider !== undefined) {
            mysqlTreeDataProvider.refresh(databaseNode);
        }
    }´

Reply:
You could pass `databaseNode` to constructor of `tableNode`

Reply:
Alright ill look into that